library/std/src/sys/uefi/alloc.rs

```diff
@@ -0,0 +1,99 @@
+//! Global Allocator for UEFI.
+//! Uses `EFI_BOOT_SERVICES.AllocatePool()` and `EFI_BOOT_SERVICES.FreePool()`.
+//! Takes a lot of inspiration from Windows allocator for Alignment > 8.
+
+use crate::alloc::{GlobalAlloc, Layout, System};
+use crate::os::uefi;
+
+pub(crate) const POOL_ALIGNMENT: usize = 8;
+// FIXME: Maybe allow changing the MEMORY_TYPE. However, since allocation is done even before main,
```

Comment:
Users who have need for such things can implement their own `#[global_allocator]`, so almost certainly std should not expose something like that.